Sources/TCPError.swift

```diff
@@ -22,16 +22,18 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-public enum TCPError: ErrorProtocol {
+import struct C7.Data
+
+public enum TCPError: Error {
     case didSendDataWithError(error: SystemError, remaining: Data)
     case didReceiveDataWithError(error: SystemError, received: Data)
 }
 
 extension TCPError: CustomStringConvertible {
     public var description: String {
         switch self {
-        case didSendDataWithError(let error, _): return "\(error)"
-        case didReceiveDataWithError(let error, _): return "\(error)"
+        case .didSendDataWithError(let error, _): return "\(error)"
```

Comment:
while we're add it, might as well update these to the swift 3 naming guidelines (no `withError`)